src/Microsoft.AspNetCore.Hosting/GenericHost/GenericWebHostBuilder.cs

```diff
@@ -0,0 +1,385 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.ExceptionServices;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.ObjectPool;
+
+namespace Microsoft.AspNetCore.Hosting.Internal
+{
+    internal class GenericWebHostBuilder : IWebHostBuilder, ISupportsStartup, ISupportsUseDefaultServiceProvider
```

Comment:
Do you have any actionable feedback? Do you see a leak?

Reply:
It does leak in that it lives beyond the obvious lifespan from the extension method, but since there's only one that shouldn't cause too much trouble. I don't have a specific suggestion here, I was only calling this out as an unusual side-effect of this design approach.